apps/src/applab/designElements/library.js

```diff
@@ -56,5 +61,40 @@ module.exports = {
     element.style.top = top + 'px';
 
     return element;
+  },
+
+  getElementPropertyTable: function (elementType) {
+    return elements[elementType].PropertyTable;
+  },
+
+  /**
+   * @param {HTMLElement} element
+   * @returns {string} String representing elementType
+   */
+  getElementType: function (element) {
+    var tagname = element.tagName.toLowerCase();
+    switch (tagname) {
+      case 'button':
+        return 'BUTTON';
```

Comment:
"errors if you have a typo" is the main benefit I was thinking of. You are correct that JS doesn't enforce this, but as a small benefit it turns out that RubyMine highlights (or grays) ElementType.BUTON with an Unresolved Variable warning.